Documentation/NvimConfiguration.md

```diff
@@ -0,0 +1,315 @@
+-Iserenity_root/Userland
+-Iserenity_root/Build/x86_64
+-Iserenity_root/Build/x86_64/Userland/Services
+-Iserenity_root/Build/x86_64/Userland/Libraries
+-Iserenity_root/Build/x86_64/Userland
+
+-DENABLE_COMPILETIME_FORMAT_CHECK
+-DSANITIZE_PTRS
+
+-pie
+-fpic
+-fdiagnostics-color=always
+
+-Wall
+-Wextra
+-Werror
+-Wno-address-of-packed-member
+-Wcast-qual
+-Wdeprecated-copy
+-Wduplicated-cond
+-Wformat=2
+-Wimplicit-fallthrough
+-Wlogical-op
+-Wmisleading-indentation
+-Wmissing-declarations
+-Wnon-virtual-dtor
+-Wsuggest-override
+-Wno-unknown-warning-option
+-Wundef
+-Wunused
+-Wno-unused-command-line-argument
+-Wwrite-strings
+
+-fdiagnostics-color=always
+-fno-delete-null-pointer-checks
+-ffile-prefix-map=serenity_root=.
+-fno-exceptions
+-fno-semantic-interposition
+-fsized-deallocation
+-fstack-clash-protection
+-fstack-protector-strong
+-g1
+
+-static-pie
+-Wno-user-defined-litereals
+-Wno-atomic-alignment
+-Wno-implicit-const-int-float-conversion
+-Wno-unused-const-variable
+-Wno-unused-private-field
+-fconstexpr-steps=16777216
+```
+
+And for the `Kernel` Directory use:
+```
+clang++
+-std=c++2a
+
+-DAK_OS_SERENITY
+-D__serenity__
+-D__unix__
+-UNO_TLS
+-Iserenity_root/Toolchain/Local/i686/i686-pc-serenity/include/c++/12.1.0
+-Iserenity_root/Toolchain/Local/i686/i686-pc-serenity/include/c++/12.1.0/i686-pc-serenity
+
+-Iserenity_root/.
+-Iserenity_root/Userland/Libraries
+-Iserenity_root/Userland/Libraries/LibC
+-Iserenity_root/Userland/Libraries/LibCrypt
+-Iserenity_root/Userland/Libraries/LibSystem
+-Iserenity_root/Userland/Services
+-Iserenity_root/Userland
+-Iserenity_root/Build/x86_64
+-Iserenity_root/Build/x86_64/Userland/Services
+-Iserenity_root/Build/x86_64/Userland/Libraries
+-Iserenity_root/Build/x86_64/Userland
+
+-DENABLE_COMPILETIME_FORMAT_CHECK
+-DSANITIZE_PTRS
+
+-pie
+-fpic
+-static-pie
+-fdiagnostics-color=always
+
+-Wall
+-Wextra
+-Werror
+-Wno-address-of-packed-member
+-Wcast-qual
+-Wdeprecated-copy
+-Wduplicated-cond
+-Wformat=2
+-Wimplicit-fallthrough
+-Wlogical-op
+-Wmisleading-indentation
+-Wmissing-declarations
+-Wnon-virtual-dtor
+-Wsuggest-override
+-Wno-unknown-warning-option
+-Wundef
+-Wunused
+-Wno-unused-command-line-argument
+-Wwrite-strings
+
+-fdiagnostics-color=always
+-fno-delete-null-pointer-checks
+-ffile-prefix-map=serenity_root=.
+-fno-exceptions
+-fno-semantic-interposition
+-fsized-deallocation
+-fstack-clash-protection
+-fstack-protector-strong
+-g1
+
+-Wno-user-defined-litereals
+-Wno-atomic-alignment
+-Wno-implicit-const-int-float-conversion
+-Wno-unused-const-variable
+-Wno-unused-private-field
+-fconstexpr-steps=16777216
+
+-DKERNEL
+-O2
+-fsigned-char
+-Wvla
+-Wnull-dereference
+-fno-rtti
+-ffreestanding
+-fbuiltin
+-mno-80387
+-mno-mmx
+-mno-sse
+-mno-sse2
+-fno-asynchronous-unwind-tables
+-nostdlib
+-ftrivial-auto-var-init=pattern
+-mcmodel=large
+-mno-red-zone
+-faligned-new=8
+-fsanitize=undefined
+```
+
+Note: Replace `serenity_root` with your root directory of the serenity os.
```

Comment:
```suggestion
Note: Replace `serenity_root` with your SerenityOS directory.
```

Reply:
Applied.

Thanks for the reviews btw :^)